app/Helpers/database/leaderboard.php

```diff
@@ -174,40 +174,43 @@ function GetLeaderboardData(
         }
     }
 
-    // Now get entries:
-    $index = $rank = $offset + 1;
-    $rankScore = null;
-    $userFound = false;
-    $entries = $leaderboard->sortedEntries()->with('user')->skip($offset)->take($numToFetch);
-    foreach ($entries->get() as $entry) {
-        if ($entry->score !== $rankScore) {
-            if ($rankScore === null) {
-                $rank = $leaderboard->getRank($entry->score);
-            } else {
-                $rank = $index;
+    if ($numToFetch != 0) {
```

Comment:
Would it be possible to early return instead of having a large conditional? eg:

```php
// untested

if ($numToFetch === 0) {
    return $retVal;
}
```